### PR TITLE
contact-us route backend

### LINKED ISSUE #33: Contatct-us route
# Backend route:

#### contact  Route
 Send a message to the site's owner.

#### Method : POST 

---

#### Main Things : 
- [x] Database
- [x] Response
- [x] Test

---

#### Database :
Use database query to insert new data related to contact. 

---
#### Response : 
Response may be success or error :
* Error :
```javascript
    {
        status:500,
        messgae:'server error'
    }
```
* Success
```javascript
    {
        status:200,
        data: `message sent successfully` , 
    }
```
---
#### Test : 
* Route : 
  Test route by testing all the expected cases for correct retrieve for data from application table.
  
* Query : 
  Test select database query to make sure that everything is ok.

### PR DESCRIPTION
**Pull request by imansedky5**
fix #33 

- added new rule   `object-curly-newline` in **.eslintrc.json**
- changed boomify parameter name from **errorMessage** into **message** in b**oomify.js**
- created contact-us route
- created validation schema (**contactusValidation**) for contact-us route
- created tests for /`api/v1/contact-us`
- modified status code in **bad request** case (400) instead of 404 in contact-us controller

### CODE REVIEW

File: server/controller/user/contactus.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const { postMessage } = require('../../database/queries');
+const { contactusValidation, boomify } = require('../../utilis');
+
+const contactusHandler = async (req, res, next) => {
+  try {
+    const { name, email, message, mobile } = req.body;
+    try {
```
Comment (mFlaifel) [inline server/controller/user/contactus.js:7]: don't use try-catch block inside another try-catch block

**Review by mFlaifel — APPROVED** [review]
'contactUs' is a common route so move it to the common folder

File: server/controller/common/contactus.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    // try {
+    const value = await contactusValidation.validateAsync({
+      name,
+      email,
+      message,
+      mobile,
+    });
+    // const { error, value } = contactusValidation.validate({
+    //   name,
+    //   email,
```
Comment (yasmeenattallah) [inline server/controller/common/contactus.js:16]: delete commented line

**Review by AdhamHaisami — DISMISSED** [review]

File: server/controller/common/contactus.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    const {
+      name, email, message, mobile,
+    } = req.body;
+    await contactusValidation.validateAsync(req.body);
+
+    await postMessage(name, email, message, mobile);
+
+    res.json({
+      status: 200,
+      message: 'message sent succesfully',
```
Comment (mFlaifel) [inline server/controller/common/contactus.js:15]: pay attention to typos
```suggestion
      message: 'message sent successfully,
```

File: __test__/contactus.test.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      body: { status, message },
+    } = await request(app).post('/api/v1/contact-us').send({
+      name: 'iman',
+      message: 'hi, i want to reserve in you salon',
+      mobile: '1235455',
+      email: 'iman@gmail.com',
+    });
+
+    expect(status).toBe(200);
+    expect(message).toBe('message sent succesfully');
```
Comment (mFlaifel) [inline __test__/contactus.test.js:24]:
```suggestion
    expect(message).toBe('message sent successfully');
```

**Review by mFlaifel — CHANGES_REQUESTED** [review]
in the database we have created_at filed 
```SQL
CREATE TABLE contacts (
  "created_at" timestamp, 
);
```
you have to insert a date into it or give it  a default value
```
"created_at" timestamp DEFAULT CURRENT_TIMESTAMP,
```

**Comment by imansedky5** [comment]
> in the database we have created_at filed
> 
> ```sql
> CREATE TABLE contacts (
>   "created_at" timestamp, 
> );
> ```
> 
> you have to insert a date into it or give it a default value
> 
> ```
> "created_at" timestamp DEFAULT CURRENT_TIMESTAMP,
> ```

done

**Review by mFlaifel — CHANGES_REQUESTED** [review]
please solve the conflict and commented test

**Comment by imansedky5** [comment]
> in the database we have created_at filed
> 
> ```sql
> CREATE TABLE contacts (
>   "created_at" timestamp, 
> );
> ```
> 
> you have to insert a date into it or give it a default value
> 
> ```
> "created_at" timestamp DEFAULT CURRENT_TIMESTAMP,
> ```

**Review by yasmeenattallah — DISMISSED** [review]

**Pull request merged**